src/main/java/com/example/springboot/service/RoleService.java

```diff
@@ -1,9 +1,10 @@
 package com.example.springboot.service;
 
 import com.example.springboot.model.Role;
+import com.example.springboot.model.RoleType;
 
 public interface RoleService {
-    Role getByName(String name);
+    Role getByName(RoleType roleType);
```

Comment:
you should not change this method - you should pass String to the service level, and here it should be transformed to RoleType and passed to the dao layer